src/job_cache/job_cache.cpp

```diff
@@ -167,9 +167,15 @@ wcl::optional<wcl::unique_fd> try_connect(std::string dir) {
   addr.sun_family = AF_UNIX;
   addr.sun_path[0] = '\0';
 
+  // Among the failures that can occur here, we could get EAGAIN. We can't use
+  // epoll to avoid this however if we get EAGAIN it means that we don't have
+  // to launch another process because the address is already bound.
   wcl::log::info("key = %s, sizeof(key) = %lu", key, sizeof(key))();
   memcpy(addr.sun_path + 1, key, sizeof(key));
   if (connect(socket_fd.get(), reinterpret_cast<const sockaddr *>(&addr), sizeof(key)) == -1) {
+    if (errno == EAGAIN) {
+      addr_bound = true;
```

Comment:
The logic around the `bool&` here is tricky and hard to following. You should just change the return type to `wcl::result` with an error enum of `enum { Generic, AddrBound }` which will make this function clearer but more importantly will make the callsite several times better

Reply:
Yeah that does seem better. I was thinking of doing something like that but I got lazy lol